src/vm/codeman.cpp

```diff
@@ -4357,6 +4357,31 @@ BOOL ExecutionManager::IsManagedCodeWorker(PCODE currentPC)
     return FALSE;
 }
 
+//**************************************************************************
+// Assumes that it is safe not to take it the ExecutionManager reader/writer lock
```

Comment:
The IsManagedCode has a version that takes a lock, but the exception handling code doesn't use it, so I just added a IsReadyToRunCode version that assumes it's not safe to take the lock. Not sure if need one with lock taking. I don't know what would use it